cmd/workflows.go

```diff
@@ -219,50 +216,85 @@ func generateServiceAccounts(namespace *corev1.Namespace, roleBindingLister rbac
 		return nil, err
 	}
 
-	// Service account for UI access
+	// Setup the default service account
 	serviceAccounts = append(serviceAccounts, &corev1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        fmt.Sprintf("argo-workflows-%v", namespace.Name),
-			Namespace:   workflowsSystemNamespace,
-			Annotations: map[string]string{"workflows.argoproj.io/rbac-rule": fmt.Sprintf("'%s' in groups", roleBinding.Subjects[0].Name)},
+			Name:      "user-default-login",
+			Namespace: namespace.Name,
+			Annotations: map[string]string{
+				"workflows.argoproj.io/rbac-rule": "true",
+				"workflows.argoproj.io/rbac-rule-precedence": "0",
+			},
 		},
 	})
 
-	// Service account for workflows
+	// The service account for argo-workflows itself
 	serviceAccounts = append(serviceAccounts, &corev1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
```

Comment:
I might recommend expanding the comment to say that it's the service account that the workflow pods will be attached to.